### PR TITLE
feat: add optional firebase environment input

### PR DESCRIPTION
**Pull request by stdavis**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -70,8 +73,22 @@ runs:
...
+        path: ./.firebase
+        key: ${{ runner.OS }}-firebase-${{ hashFiles('**/*.cache') }}
+        restore-keys: |
+          ${{ runner.OS }}-firebase-
+          ${{ runner.OS }}-
+
+    - name: ♻️ Set Firebase environment
+      if: inputs.environment != ''
+      shell: bash
+      run: npx firebase-tools use ${{ inputs.environment }}
```
Comment (steveoh) [inline action.yml:88]: isn't this equivalent to using the --project flag like we use in the deploy and as an input? 

https://firebase.google.com/docs/cli/#:~:text=You%20can%20override,project%3Dprod.

Would we be able to update the value of `${{ inputs.project-id }}` to deploy to an environment?

**Review by steveoh — COMMENTED** [review]
The docs make this change not sound necessary but i'll trust your judgement after you read my review comment.

File: action.yml [hunk 2]
```diff
@@ -38,6 +38,9 @@ inputs:
   service-now-password:
     description: the service account password
     required: false
+  environment:
+    description: the firebase environment to use (https://firebase.google.com/docs/functions/config-env)
+    required: false
```
Comment (steveoh) [inline action.yml:43]: **thought**: this could be merged with project-id's description

Comment (stdavis) [reply action.yml:43]: Oops. I forgot to remove this.

**Comment by stdavis** [comment]
> Since we're in here do we want to try pushing the deploy result to the GitHub action annotations?

I'd like to move on with electrofishing.

**Review by steveoh — APPROVED** [review]
Here we go!

**Pull request merged**